src/chain/index.ts

```diff
@@ -73,21 +84,28 @@ export async function verifyProofOnChain(proof: Proof) {
       ],
     ],
     [BigInt(proofData.pi_c[0]), BigInt(proofData.pi_c[1])],
+    // TODO: this is parsed when getting the data from the backend,
+    // add propper typing from the start
+    // @ts-ignore
     proof.props.publicOutputs.map((output) => BigInt(output)),
   ] as const;
 
-  console.log("Call data to verify: ", args);
-  console.log("contract address: ", proof.blueprint.props.verifierContract.address);
+  let calldata = await snarkjs.groth16.exportSolidityCallData(proofData, proof.props.publicOutputs);
+  calldata = JSON.parse(`[${calldata}]`);
 
   try {
     await client.readContract({
       address: proof.blueprint.props.verifierContract.address as `0x${string}`,
+      // TODO: this is parsed when getting the data from the backend,
+      // add propper typing from the start
+      // @ts-ignore
       abi: getVerifierContractAbi(proof.props.publicOutputs.length),
       functionName: "verify",
-      args,
+      args: [1, ...calldata],
```

Comment:
using a ProofType enum here would be better
```suggestion
      args: [ProofType.Server, ...calldata],
```